exotations/dynamics_solvers/exotica_cartpole_dynamics_solver/src/cartpole_dynamics_solver.cpp

```diff
@@ -0,0 +1,141 @@
+        ThrowPretty("Robot model may not be a Cartpole.");
+}
+
+Eigen::VectorXd CartpoleDynamicsSolver::f(const StateVector& x, const ControlVector& u)
+{
+    auto theta = x(1);
+    auto xdot = x(2);
+    auto thetadot = x(3);
+
+    auto l = this->l;
+    auto m_c = this->m_c;
+    auto m_p = this->m_p;
+    auto g = this->g_;
+    auto s = sin(theta);
+    auto c = cos(theta);
+    auto tdots = thetadot * thetadot;
+    auto ss = s * s;
+
+    auto x_dot = StateVector(4);
+    x_dot << xdot, thetadot,
+        (u(0) + m_p * s * (l * tdots + g * c)) /
+            (m_c + m_p * ss),
+        -(l * m_p * c * s * tdots + u(0) * c + (m_c + m_p) * g * s) /
+            (l * m_c + l * m_p * ss);
+
+    return x_dot;
+}
+
+Eigen::MatrixXd CartpoleDynamicsSolver::fx(const StateVector& x, const ControlVector& u)
+{
+    auto theta = x(1);
+    auto xdot = x(2);
+    auto tdot = x(3);
+
+    auto l = this->l;
+    auto m_c = this->m_c;
+    auto m_p = this->m_p;
+    auto g = this->g_;
+    auto s = sin(theta);
+    auto c = cos(theta);
+    auto ss = s * s;
+
+    Eigen::Matrix4d fx;
+    fx << 0, 0, 1, 0,
+        0, 0, 0, 1,
+        //
+        0,
+        -2 * m_p * (m_p * (g * cos(theta) + l * tdot * tdot) * sin(theta) + u(0)) * sin(theta) * cos(theta) / pow(m_c + m_p * sin(theta) * sin(theta), 2) + (-g * m_p * sin(theta) * sin(theta) + m_p * (g * cos(theta) + l * tdot * tdot) * cos(theta)) / (m_c + m_p * sin(theta) * sin(theta)),
+        0,
+        2 * l * m_p * tdot * sin(theta) / (m_c + m_p * sin(theta) * sin(theta)),
+        //
+        0,
+        -2 * l * m_p * (-g * (m_c + m_p) * sin(theta) - l * m_p * tdot * tdot * sin(theta) * cos(theta) - u(0) * cos(theta)) * sin(theta) * cos(theta) / pow(l * m_c + l * m_p * sin(theta) * sin(theta), 2) + (-g * (m_c + m_p) * cos(theta) + l * m_p * tdot * tdot * sin(theta) * sin(theta) - l * m_p * tdot * tdot * cos(theta) * cos(theta) + u(0) * sin(theta)) / (l * m_c + l * m_p * sin(theta) * sin(theta)),
+        0,
+        -2 * l * m_p * tdot * sin(theta) * cos(theta) / (l * m_c + l * m_p * sin(theta) * sin(theta));
+
+    return fx;
+}
+
+Eigen::MatrixXd CartpoleDynamicsSolver::fu(const StateVector& x, const ControlVector& u)
+{
+    auto theta = x(1);
+    auto xdot = x(2);
+    auto tdot = x(3);
+
+    auto l = this->l;
+    auto m_c = this->m_c;
+    auto m_p = this->m_p;
+    auto g = this->g_;
+    auto s = sin(theta);
+    auto c = cos(theta);
+    auto ss = s * s;
+
+    Eigen::Vector4d fu;
+    fu << 0, 0, 1 / (m_c + m_p * sin(theta) * sin(theta)), -cos(theta) / (l * m_c + l * m_p * sin(theta) * sin(theta));
+    return fu;
+}
+
+Eigen::VectorXd CartpoleDynamicsSolver::GetPosition(Eigen::VectorXdRefConst x_in)
+{
+    auto joint_pos = Eigen::Vector2d();
+    joint_pos << x_in(0), M_PI - x_in(1);
+    return joint_pos;
+}
+
+Eigen::VectorXd CartpoleDynamicsSolver::StateDelta(const StateVector& x_1, const StateVector& x_2)
+{
+    Eigen::VectorXd diff = x_1 - x_2;
```

Comment:
Have you tried it without this since fixing the other bug?

Reply:
Forgot about that. It's exactly the same.

Reply:
Cool, let's remove the override then

Reply:
That looks done?